### PR TITLE
Sql test record2022.05

### LINKED ISSUE #28621: Re-record SQL tests
The Azure.Resourcemanager.Sql SDK has updated API versions for some resources but the tests are not re-recorded. It's not caught in CI as the tests are marked with `RunTestFrequency.Manually` and skipped in CI tests. We should re-record the tests before GA.

### PR DESCRIPTION
**Pull request by mcgallan**
Completed Sql Test re-record

### CODE REVIEW

**Review by Yao725 — DISMISSED** [review]
Nice work, but I think we should remove the attribute `RunFrequency` in `SqlManagementClientBase.cs`, otherwise the tests will not run in the pipeline.

**Comment by mcgallan** [comment]
@fengzhou-msft I made some changes to this part of the Sql test, on the one hand, because the sql test takes too much time to run, so I implemented playback passing by modifying the json file in SessionRecords, and on the other hand, in the SqlManagementClientBase .cs，The original code 
image
caused an error to be reported，because
image
So I changed SourceType="Microsoft.Sql" to SourceType="/Microsoft.Sql" to ensure that the test case passed, but the type in the json file also changed
image
I wasn't sure if there was a more appropriate way to modify it, so I left this comment

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/tests/Scenario/ManagedInstanceTests.cs [hunk 1]
```diff
@@ -62,7 +62,7 @@ public async Task GlobalSetUp()
                         AddressPrefix = "10.10.2.0/24",
                         Delegations =
                         {
-                            new ServiceDelegation() { ServiceName  = "Microsoft.Sql/managedInstances",Name="Microsoft.Sql/managedInstances" ,ResourceType="Microsoft.Sql"}
+                            new ServiceDelegation() { ServiceName  = "Microsoft.Sql/managedInstances",Name="Microsoft.Sql/managedInstances" ,ResourceType="/Microsoft.Sql"}
```
Comment (Yao725) [inline sdk/sqlmanagement/Azure.ResourceManager.Sql/tests/Scenario/ManagedInstanceTests.cs:65]: Is the slash used to bypass the `ResourceType` validation? Based on @ArthurMa1978's investigation, can we change this property value to `Microsoft.Sql/managedInstances`?

**Review by Yao725 — COMMENTED** [review]
Could you also please resolve the conflicts?

File: sdk/sqlmanagement/Azure.ResourceManager.Sql/tests/Scenario/ManagedInstanceTests.cs (hunk 1 above)
Comment (mcgallan) [reply sdk/sqlmanagement/Azure.ResourceManager.Sql/tests/Scenario/ManagedInstanceTests.cs]: All the above modifications are complete

**Review by fengzhou-msft — APPROVED** [review]

**Review by Yao725 — APPROVED** [review]

**Pull request merged**